src/api/spec/services/user_service/involved_spec.rb

```diff
@@ -0,0 +1,118 @@
+require 'rails_helper'
+
+RSpec.describe UserService::Involved, vcr: true do
+  let(:user) { create(:confirmed_user, :with_home) }
+  let(:home_project) { user.home_project }
+
+  let(:some_project) { create(:project_with_package) }
+  let(:some_package) { some_project.packages.first }
+
+  let(:another_project) { create(:project, name: 'OBS:Unstable') }
+
+  let(:non_involved_project) { create(:project_with_package) }
+  let(:non_involved_package) { some_other_project.packages.first }
+
+  let!(:project_relationship) { create(:relationship_project_user, user: user, project: some_project) }
+  let!(:package_relationship) do
+    create(:relationship_package_user,
+           user: user,
+           package: some_package,
+           role: Role.find_by_title!('bugowner'))
+  end
+
+  let!(:another_project_relationship) do
+    create(:relationship_project_user,
+           user: user,
+           project: another_project,
+           role: Role.find_by_title!('bugowner'))
+  end
+
+  let(:involved_service) { described_class.new(user: user, filters: filters, page: nil) }
+
+  before do
+    User.session = user
+  end
+
+  describe '#involved_packages_and_projects' do
+    context 'when there are no filters set' do
+      let(:filters) { { 'search_text' => '' } }
+
+      subject { involved_service.involved_pkg_and_prj_paginated }
```

Comment:
This `subject` line, as it is the same in all the `context` clauses and all the only `describe` section, could be moved outside, just after the `before` clause in line 32. And the rest `subject` lines, could be removed.

Reply:
done